netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -247,9 +254,18 @@ private NettyTransportFactory(Class<? extends Channel> channelType,
 
     @Override
     public ClientTransport newClientTransport(SocketAddress serverAddress, String authority) {
-      GrpcUtil.checkAuthority(authority);
-      return new NettyClientTransport(serverAddress, channelType, group,
-          createProtocolNegotiator(authority, negotiationType, sslContext),
+      String host = "";
+      int port = -1;
+      // Don't check the authority again here, because it will have already been validated by the
+      // parent class.
+      if (serverAddress instanceof InetSocketAddress) {
```

Comment:
I picked this because I didn't see where the host and port were originally set from.  The forAddress(SocketAddress serverAddress) constructor doesn't give the host and port connecting to, so I don't know where to get them from.